src/beliefpropagation.jl

```diff
@@ -0,0 +1,244 @@
+#Construct the random initial Message Tensors for an ITensor Network, based on a partitioning into subgraphs specified ny 'sub graphs'
+#The ITensorNetwork needs to be flat (i.e. just sites and link indices, no site indices), and is assumed to only have 1 link indice between any two sites
+#id_init = 0 => Random, id_init = 1 => Identity Matrix initialisation (preferred)
+function construct_initial_mts(
+  g::NamedDimGraph, flatpsi::ITensorNetwork, dg_subgraphs::DataGraph; id_init=1
```

Comment:
An alternative syntax for choosing the tensor initialization that would allow more flexibility would be the following:
```julia
construct_initial_mts(g, tn, subgraphs; init=ones) # Constant
construct_initial_mts(g, tn, subgraphs; init=randn) # Normal distribution around 0
construct_initial_mts(g, tn, subgraphs; init=rand) # Uniform distribution from 0 to 1
construct_initial_mts(g, tn, subgraphs; init=(d -> 2rand(d) .- 1) # Uniform distribution from -1 to 1
```
Then, internally you can just construct the tensors with:
```julia
itensor(init(dim(edge_inds)), edge_inds)
```

(This makes me think that we should have ITensor/ITensorNetwork constructors like this that can take an arbitrary function to allow flexibility in initializing tensors various distributions but that's a separate issue.)

Reply:
Ah yes this is nice. I did change this but the only difficulty is I can't find a shorthand for the identity tensor? I.e. there seems to be no Julianic way of doing eye(dim(edge_inds)) and getting an identity tensor? Hence I had to add in a separate if statement so if someone puts init="I" it will build an identity tensor. Do you know of a way around this?

Reply:
Hmm good question, I think I was confused about what type of tensor you were trying to make (I thought you were trying to make a tensor filled with all ones but I see you want to make a diagonal tensor with diagonal values of one).

A more flexible interface would be the following:
```julia
construct_initial_mts(g, tn, subgraphs; init=(I...) -> allequal(I) ? 1 : 0) # Diagonal tensor
construct_initial_mts(g, tn, subgraphs; init=(I...) -> 1) # Tensor filled uniformly with 1
construct_initial_mts(g, tn, subgraphs; init=(I...) -> randn()) # Normal distribution around 0
construct_initial_mts(g, tn, subgraphs; init=(I...) -> rand()) # Uniform distribution from 0 to 1
construct_initial_mts(g, tn, subgraphs; init=(I...) -> 2rand() - 1) # Uniform distribution from -1 to 1
```
so it would take a function that maps an index of the tensor to a value. Then the tensor could be constructed internally like:
```julia
itensor([init(Tuple(I)...) for I in CartesianIndices(tuple(dim.(edge_inds)...))], edge_inds)
```

Reply:
Okay great, I have designed it that way now.